pkg/client/add_events_test.go

```diff
@@ -634,6 +667,314 @@ func TestAddEventsAreRejectedOncePreviousReqRetriesMaxLifetimeNotExpired(t *test
 	assert.Errorf(t, err, "AddEvents - reject batch: rejecting - Last HTTP request contains an error: failed to handle previous request")
 }
 
+func TestAddEventsServerHostLogic(t *testing.T) {
+	configServerHost := "global-server-host"
+	ev1ServerHost := "host-1"
+	ev2ServerHost := "host-2"
+	ev3ServerHost := "host-3"
+	ev4ServerHost := "host-4"
+	ev5ServerHost := "host-5"
+	key := attributeKey
+	ev1Value := "event-1-value"
+	ev2Value := "event-2-value"
+	ev3Value := "event-3-value"
+	ev4Value := "event-4-value"
+	ev5Value := "event-5-value"
+
+	tests := []struct {
+		name     string
+		events   []tEvent
+		groupBy  []string
+		expCalls [][]tAttr
+	}{
+		// when nothing is specified, there is just once call
+		{
+			name: "no server host is specified",
+			events: []tEvent{
+				{
+					attrs: tAttr{key: ev1Value},
+				},
+				{
+					attrs: tAttr{key: ev2Value},
+				},
+			},
+			expCalls: [][]tAttr{
+				{
+					{key: ev1Value, add_events.AttrOrigServerHost: configServerHost},
+					{key: ev2Value, add_events.AttrOrigServerHost: configServerHost},
+				},
+			},
+		},
+
+		// when serverHost is specified and is same as global one, there is just once call
+		{
+			name: "serverHost is same as global",
+			events: []tEvent{
+				{
+					attrs:      tAttr{key: ev1Value},
+					serverHost: configServerHost,
+				},
+				{
+					attrs: tAttr{key: ev2Value},
+				},
+			},
+			expCalls: [][]tAttr{
+				{
+					{key: ev1Value, add_events.AttrOrigServerHost: configServerHost},
+					{key: ev2Value, add_events.AttrOrigServerHost: configServerHost},
+				},
+			},
+		},
+
+		// when serverHost is specified and is different from global one, there are two calls
+		{
+			name: "serverHost is different from global",
+			events: []tEvent{
+				{
+					attrs:      tAttr{key: ev1Value},
+					serverHost: ev1ServerHost,
+				},
+				{
+					attrs: tAttr{key: ev2Value},
+				},
+			},
+			expCalls: [][]tAttr{
+				{
+					{key: ev1Value, add_events.AttrOrigServerHost: ev1ServerHost},
+					{key: ev2Value, add_events.AttrOrigServerHost: configServerHost},
+				},
+			},
+		},
+
+		// when serverHost is specified and is same as attribute serverHost
+		{
+			name: "serverHost and attribute serverHost are same",
+			events: []tEvent{
+				{
+					attrs:      tAttr{key: ev1Value, add_events.AttrServerHost: ev1ServerHost},
+					serverHost: ev1ServerHost,
+				},
+				{
+					attrs: tAttr{key: ev2Value},
+				},
+			},
+			expCalls: [][]tAttr{
+				{
+					{key: ev1Value, add_events.AttrOrigServerHost: ev1ServerHost},
+					{key: ev2Value, add_events.AttrOrigServerHost: configServerHost},
+				},
+			},
+		},
+
+		// when serverHost is specified and is same as attribute serverHost then attribute serverHost wins
+		{
+			name: "serverHost and attribute serverHost differs",
+			events: []tEvent{
+				{
+					attrs:      tAttr{key: ev1Value, add_events.AttrServerHost: ev1ServerHost},
```

Comment:
I would actually imagine here `event.ServerHost` should win / have priority over one in attributes.

Reply:
To clarify further, I think `Event.ServerHost` should have priority since it's a special top level attribute for us and it should also be set by the consumer (otel exporter) in case `LogRecord.attributes` contains `serverHost`.

Reply:
Fixed